historyarchive/scan.go

```diff
@@ -377,8 +377,8 @@ func (arch *Archive) CheckBucketsMissing() map[Hash]bool {
 	defer arch.mutex.Unlock()
 	missing := make(map[Hash]bool)
 	for k := range arch.referencedBuckets {
-		_, ok := arch.allBuckets[k]
-		if !ok {
+		bucketExists, mapEntryExists := arch.allBuckets[k]
+		if !bucketExists || !mapEntryExists {
```

Comment:
nit: `allBuckets` is `map[Hash]bool` so we can exploit a default value:
```suggestion
		bucketExists := arch.allBuckets[k]
		if !bucketExists {
```